### PR TITLE
feat: setting custom resource directories in Gradle

### PR DESCRIPTION
**Pull request by zamulla**

### CODE REVIEW

File: topics/compose/compose-images-resources.md [hunk 1]
```diff
@@ -63,6 +68,43 @@ compose.resources {
...
+}
+```
+
+You can also set up a folder populated by a Gradle task, for example, with [downloaded files](#remote-files):
+
+```kotlin
+abstract class DownloadRemoteFiles : DefaultTask() {
+    @get:Inject
+    abstract val layout: ProjectLayout
+
```
Comment (terrakok) [inline topics/compose/compose-images-resources.md:91]: delete

File: topics/compose/compose-images-resources.md [hunk 2]
```diff
@@ -63,6 +68,43 @@ compose.resources {
...
+    @get:OutputDirectory
+    val outputDir = layout.buildDirectory.dir("downloadedRemoteFiles")
+
+    @TaskAction
+    fun run() { /* your code for downloading files */ }
+}
+
+compose.resources {
+    customDirectory(
+            sourceSetName = "jvmMain",
```
Comment (terrakok) [inline topics/compose/compose-images-resources.md:101]: better: `sourceSetName = "iosMain",`

**Review by terrakok — CHANGES_REQUESTED** [review]

File: topics/compose/compose-images-resources.md [hunk 3]
```diff
@@ -45,13 +45,18 @@ To access resources in your multiplatform projects:
 ## Configuration
 
 You can alter the default settings for Compose Multiplatform resources by adding the `compose.resources {}` block
-to the `build.gradle.kts` file:
+to the `build.gradle.kts` file.
+
+### Class generation settings
+
+There are several settings that affect the way the `Res` class is generated for your project.
+An example configuration can look like this:
```
Comment (AlejandraPedroza) [inline topics/compose/compose-images-resources.md:53]:
```suggestion
An example configuration looks like this:
```

File: topics/compose/compose-images-resources.md [hunk 4]
```diff
@@ -63,6 +68,41 @@ compose.resources {
   when the resource library is only available transitively. By default, the `auto` value is used, to generate the `Res`
```
Comment (AlejandraPedroza) [inline topics/compose/compose-images-resources.md:68]:
```suggestion
  when the resource library is only available transitively. By default, the `auto` value is used to generate the `Res`
```

File: topics/compose/compose-images-resources.md [hunk 5]
```diff
@@ -63,6 +68,41 @@ compose.resources {
...
+    @get:OutputDirectory
+    val outputDir = layout.buildDirectory.dir("downloadedRemoteFiles")
+
+    @TaskAction
+    fun run() { /* your code for downloading files */ }
+}
+
+compose.resources {
+    customDirectory(
+            sourceSetName = "iosMain",
```
Comment (AlejandraPedroza) [inline topics/compose/compose-images-resources.md:99]: Is this indentation correct?

**Review by AlejandraPedroza — COMMENTED** [review]
Look good! Just minor comments to consider.

File: topics/compose/compose-images-resources.md (hunk 5 above)
Comment (zamulla) [reply topics/compose/compose-images-resources.md:99]: Nope!

File: topics/compose/compose-images-resources.md [hunk 6]
```diff
@@ -63,6 +68,41 @@ compose.resources {
...
+```kotlin
+compose.resources {
+    customDirectory(
+        sourceSetName = "desktopMain",
+        directoryProvider = provider { layout.projectDirectory.dir("desktopResources") }
+    )
+}
+```
+
+You can also set up a folder populated by a Gradle task, for example, with [downloaded files](#remote-files):
```
[the remark this answers is not in the document]
Comment (zamulla) [reply topics/compose/compose-images-resources.md]: I think I got confused here, removed the link now.

**Review by AlejandraPedroza — APPROVED** [review]
LGTM 👍

**Pull request merged**